libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/RoutesRenderedCallback.kt

```diff
@@ -0,0 +1,82 @@
+package com.mapbox.navigation.ui.maps.route.line.api
+
+import com.mapbox.maps.MapboxMap
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+
+/**
+ Callback to be notified of rendered routes.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+interface RoutesRenderedCallback {
+
+    /**
+     * Map instance where the routes are expected to be rendered.
+     */
+    val map: MapboxMap
+
+    /**
+     * Invoked when routes are rendered.
+     *
+     * @param result [RoutesRenderedResult] describing rendering result
+     */
+    fun onRoutesRendered(result: RoutesRenderedResult)
+}
+
+/**
+ * Class describing the result of routes rendering.
+ *
+ * @param successfullyRenderedRouteIds ids of the routes that were successfully rendered (correspond to [NavigationRoute.id])
+ * @param renderingCancelledRouteIds ids of the routes whose rendering was cancelled (correspond to [NavigationRoute.id]).
+ *  This may happen, for example, when a newer rendering operation had been queued before the previous one finished.
+ * @param successfullyClearedRouteIds ids of the routes that were successfully cleared (correspond to [NavigationRoute.id])
+ * @param clearingCancelledRouteIds ids of the routes whose clearance was cancelled (correspond to [NavigationRoute.id]).
+ *  This may happen, for example, when a newer rendering operation had been queued before the previous one finished.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+class RoutesRenderedResult internal constructor(
+    val successfullyRenderedRouteIds: Set<String>,
+    val renderingCancelledRouteIds: Set<String>,
+    val successfullyClearedRouteIds: Set<String>,
+    val clearingCancelledRouteIds: Set<String>,
```

Comment:
We still need to keep track of these ids to understand when all routes are processed. And after that we can either expose that info or not. IMO for consistency we should expose it (so that the passed ids are never "lost").